src/domain/profile/modal/EditMyProfileForm.tsx

```diff
@@ -77,18 +79,34 @@ export default function EditMyProfileForm({
   const onSubmit = async (payload: UpdateMyProfileMutationInput) => {
     setFormIsFilling(false);
     try {
+      const input = {
+        firstName: payload.firstName,
+        lastName: payload.lastName,
+        phoneNumber: payload.phoneNumber,
+        language: payload.language,
+        hasAcceptedMarketing: payload.hasAcceptedMarketing,
+        languagesSpokenAtHome: payload.languagesSpokenAtHome,
+      };
       await updateMyProfile({
         variables: {
-          input: {
-            firstName: payload.firstName,
-            lastName: payload.lastName,
-            phoneNumber: payload.phoneNumber,
-            language: payload.language,
-            hasAcceptedMarketing: payload.hasAcceptedMarketing,
-            languagesSpokenAtHome: payload.languagesSpokenAtHome,
-          },
+          input,
         },
       });
+
+      updateProfile(
+        (prevValue) =>
+          ({
+            ...(prevValue ?? initialValues),
+            ...input,
+            languagesSpokenAtHome: {
+              __typename: 'LanguageNodeConnection',
+              edges:
+                input.languagesSpokenAtHome?.map((id) => ({ node: { id } })) ??
+                [],
+            },
+          }) as MyProfile
+      );
```

Comment:
It sure would be better, but I need to double check whether it is OK for the API, because then also the updateMyProfile-mutation would use those default values.

Reply:
I made some changes which based on this and the guard is now on.